### PR TITLE
[GEOS-11150] add geoserver-monitor-kafka extension

### PR DESCRIPTION
**Pull request by spike83**
[GEOS-11150](https://osgeo-org.atlassian.net/browse/GEOS-11150) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=geoserver&utm_campaign=badge_info)

If we want to analyze the data produced by the monitoring extension we had the option to write to a database using hibernate. Going forward I’d like also to be able to stream the monitoring data into a Kafka topic.

With stream processing we can build the aggregates we want e.g. with KsqlDB, Flink or Kafka Streams Application and provide reports in near real time… But foundation is to have the data streamed into a topic…

With this extension we can stream the data into a Kafka topic and configure all the producer options as we want to.
  

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geoserver.org/latest/en/developer/policies/committing.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [x] All the build checks are green ([see automated QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html)).

For core and extension modules:

- [x] New unit tests have been added covering the changes.
- [x] [Documentation](https://github.com/geoserver/geoserver/tree/main/doc/en/user/source) has been updated (if change is visible to end users).
- [ ] The [REST API docs](https://github.com/geoserver/geoserver/tree/main/doc/en/api/1.0.0) have been updated (when changing configuration objects or the REST controllers).
- [x] There is an issue in the [GeoServer Jira](https://osgeo-org.atlassian.net/browse/GEOS/summary) (except for changes that do not affect administrators or end users in any way).
- [ ] Commit message(s) must be in the form ``[GEOS-XYZWV] Title of the Jira ticket``.
- [ ] Bug fixes and small new features are presented as a single commit.
- [ ] Each commit has a single objective (if there are multiple commits, each has a separate JIRA ticket describing its goal).

### CODE REVIEW

**Comment by aaime** [comment]
Whoops, looks like it has accumulated conflicts. Personally I have no experience with Kafka, cannot review the details, but the direction looks like a good one to me. Anyone else that wants to have a look before merge?

File: src/community/pom.xml [hunk 1]
```diff
@@ -568,6 +570,25 @@
         <module>monitor-hibernate</module>
       </modules>
     </profile>
+    <profile>
+      <id>monitor-kafka</id>
+      <build>
+        <plugins>
+          <plugin>
+            <artifactId>maven-assembly-plugin</artifactId>
+            <version>3.2.0</version>
```
Comment (mprins) [inline src/community/pom.xml:579]: this version is already provided through pluginManagment of the parent pom:
https://github.com/geoserver/geoserver/blob/f5355bc54f1a09b2c429406002dddd09f2f5fac3/src/pom.xml#L1699-L1702

```suggestion
```

File: src/web/app/pom.xml [hunk 2]
```diff
@@ -629,8 +629,23 @@
       </activation>
       <dependencies>
         <dependency>
-          <groupId>org.geoserver.extension</groupId>
+          <groupId>org.geoserver.community</groupId>
           <artifactId>gs-monitor-hibernate</artifactId>
+          <version>2.25-SNAPSHOT</version>
```
Comment (mprins) [inline src/web/app/pom.xml:634]:
```suggestion
          <version>${project.version}</version>
```

**Review by mprins — CHANGES_REQUESTED** [review]
I've provided some small changes to the pom files.

Aside there seems to be Kafka 3.6.0; but since I haven't used kafka I'm not sure if using that is opportune (compared to 3.5.1)

**Comment by spike83** [comment]
@mprins thanks for your review. I updated the deps. Kafka 3.6.0 was released in the mean time. Not critical but for sure a good idea to update.

**Review by mprins — APPROVED** [review]

**Comment by aaime** [comment]
Merged! If you want to backport, that will have to be a manual PR, as you'll need a different version number in the pom.xml file (and potentially, I have not checked, some different API in GeoTools, if you want to go down to 2.23.x).

**Pull request merged**